### PR TITLE
Access control

### LINKED ISSUE #9: Access control system
Students should only be able to access their resources. Staff can probably be granted access to everything.

### PR DESCRIPTION
**Pull request by dfaulken**
Taking care of access whitelisting for students, and a few other tweaks.

Closes #9.

### CODE REVIEW

**Comment by dfaulken** [comment]
Much of this functionality is taken from round-three.

On any page that a student should be able to view (or POST to), simply write `permit_student_access`. If any student attempts to access a page for which this is not enabled, they receive the 401 Access Denied page (or for XHR requests, blank content with a 401 status).

File: app/views/dashboard/_pending_interviews.haml [hunk 1]
```diff
@@ -7,7 +7,7 @@
         %li
           = position.name
           %ul
-            - if @pending_interviews.key position
+            - if @pending_interviews.keys.include? position
```
Comment (nmonti) [inline app/views/dashboard/_pending_interviews.haml]: NITPICK: `has_key?`

**Comment by nmonti** [comment]
:+1:

File: app/views/dashboard/_pending_interviews.haml (hunk 1 above)
Comment (dfaulken) [inline app/views/dashboard/_pending_interviews.haml]: Yep, and in fact resolving the merge conflict ought to bring that in.

**Pull request merged**